starknet-foundry/crates/forge/tests/data/contracts/constructor_roll_checker.cairo

```diff
@@ -0,0 +1,26 @@
+#[starknet::interface]
+trait IConstructorRollChecker<TContractState> {
+    fn was_rolled_on_construction(ref self: TContractState, expected_block_number: u64);
+}
+
+#[starknet::contract]
+mod ConstructorRollChecker {
+    use box::BoxTrait;
+    #[storage]
+    struct Storage {
+        blk_nb: u64,
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState) {
+        let block_numb = starknet::get_block_info().unbox().block_number;
+        self.blk_nb.write(block_numb);
+    }
+
+    #[external(v0)]
+    impl IConstructorRollChecker of super::IConstructorRollChecker<ContractState> {
+        fn was_rolled_on_construction(ref self: ContractState, expected_block_number: u64) {
+            assert(self.blk_nb.read() == expected_block_number, 'block_numb incorrect');
```

Comment:
It seems unnatural to have asserts inside the test data. Maybe put it to the test function?

same for  https://github.com/foundry-rs/starknet-foundry/pull/181/files#diff-f77438c3fa2d169290375e9aea86788dfd6b6979080c69d7a8ccadda626c7c8dR18

Reply:
[Update logic to getting block number and assertions in tests](https://github.com/foundry-rs/starknet-foundry/pull/181/commits/33421c4864d0e82d0cb0eab28d84410ccc018bb8)